### DISCUSSION TITLE
Different response models in the OpenAPI spec depending on content type

### DISCUSSION THREAD

**Discussion by zbrox** [fastapi/fastapi#3430] [Questions] [answered]
Hey everyone,

I've been trying to figure out a good way to define endpoints which return different models depending on the Content-Type header. Implementing it is fairly easy but the OpenAPI spec remains unchanged since it's all in the implementation of the endpoint function.

Is there currently a way, per path, to define different response model based on media type?

**Comment by YuriiMotov** [comment, marked as answer]
You can specify different response schemas for different content types using `responses` parameter of path decorator:

```python
from fastapi import FastAPI, Request
from pydantic import BaseModel

app = FastAPI()

class Item(BaseModel):
    id: int
    name: str

@app.get(
    "/item",
    response_model=Item,
    responses={
        200: {
            "content": {
                "application/json": {"schema": Item.model_json_schema()},
                "text/plain": {"schema": {"type": "string"}},
            },
            "description": "Returns JSON or plain text based on Accept header",
        }
    },
)
async def get_item(request: Request):
    pass
```
